en/integrations/plugin-packs/procedures/hardware-devices-barco-cs-restapi.md

```diff
@@ -0,0 +1,151 @@
+
+## How to test the Plugin and what are the main options for?
+
+Once the Plugin installed, log into your Poller using the *centreon-engine* user account and test by running the following command
+(Parameters such as ```api-username``` or ```api-password```have to be adjusted):
+
+```bash
+/usr/lib/centreon/plugins/centreon_barco_cs_restapi.pl \
+    --plugin=hardware::devices::barco::cs::restapi::plugin \
+    --mode=device \
+    --hostname='10.30.2.79' \
+    --port='4001' \
+    --proto='https' \
+    --api-username='myapiusername' \
+    --api-password='myapipassword' \
+    --warning-cpu-temperature='55' \
+    --critical-cpu-temperature='60' \
+    --verbose
+```
+
+Output example:
+```
+OK: device is ok | 'cpu#hardware.temperature.celsius'=52C;0:55;0:60;; 'pcie#hardware.temperature.celsius'=50C;0:55;0:60;;
+checking device
+    status: ok
+    temperature cpu 52 C, pcie 50 C
+    process 'Button Agent' status is running
+    process 'ClickShare Server' status is running
+    process 'Config Manager' status is running
+    process 'DBus Daemon' status is running
+    process 'DHCP Server' status is running
+    process 'Device Daemon' status is running
+    process 'Graphics Server' status is running
+    process 'Job Scheduler' status is running
+    process 'LED Control' status is running
+    process 'Miracast Server' status is running
+    process 'Process Monitor' status is running
+    process 'System Logging' status is running
+    process 'WebUI Server' status is running
+    process 'Wifi Access Point Daemon' status is running
+```
+
+The command above monitors device (```--mode=device```).
+
+It uses api-username (```--api-username='myapiusername'```), an api-password (```--api-password='myapipassword'```)
+and it connects to the host _10.30.2.79_ (```--hostname='10.30.2.79'```)
+on the port 4001 (```--port='4001'```) using https (```--proto='https'```).
+
+All the options that can be used with this plugin can be found over the ```--help``` command:
+
+```bash
+/usr/lib/centreon/plugins/centreon_barco_cs_restapi.pl \
+    --plugin=hardware::devices::barco::cs::restapi::plugin \
+    --mode=device \
+    --help
+```
+
+## Troubleshooting
+
+### ```UNKNOWN: 500 Can't connect to 10.30.2.79:4001```
+
+This error message means that the Centreon Plugin couldn't successfully connect to the Barco ClickShare API.
+Check that no third party device (such as a firewall) is blocking the request.
+A proxy connection may also be necessary to connect to the API. This can be done by using the ```--proxyurl='http://proxy.mycompany:8080'``` option in the command.
+
+### ```UNKNOWN: 501 Protocol scheme 'connect' is not supported```
+
+When using a proxy to connect to the Barco ClickShare API, this error message means that the Centreon Plugin library does not support
+the proxy connection protocol.
+
+In order to prevent this issue, use the *curl* HTTP backend by adding the following option to the command: ```--http-backend='curl'```.
```

Comment:
Use a link to the dedicated troubleshooting topic. 

Maybe you can add the 501 error, I think I forgot it ;)